File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        char element;
        int read = 0;
        int write = 0;
        int n = chars.length;
        int count;
        while(read < n){
            element = chars [read];
            count = 0;
            while(read < n && chars [read] == element){
                read++;
                count++;
            }
            chars [write] = element;
            write++;
            if (count > 1){
                String strcount = String.valueOf(count);
                for(int i = 0; i < strcount.length(); i++){
                    chars [write] = strcount.charAt(i);
                    write++;
                }
            }
        }
        return write;
    }
}

File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        read = 0
        write = 0
        n = len(chars)

        while read < n:
            element = chars[read]
            count = 0

            while read < n and chars[read] == element:
                read += 1
                count += 1

            chars[write] = element
            write += 1

            if count > 1:
                strcount = str(count)
                for ch in strcount:
                    chars[write] = ch
                    write += 1

        return write